virttest/guest_agent.py

```diff
@@ -730,3 +725,86 @@ def fsthaw(self, check_status=True):
                     raise
             return ret
         return -1
+
+    @error_context.context_aware
+    def file_open(self, path, mode=None):
+        """
+        Open a guest file
+        :param path: full path to the file in the guest to open.
+        :param mode: optional open mode, "r" is the default value.
+        :return: filehandle
+        """
+        cmd = "guest-file-open"
+        if self.check_has_command(cmd):
+            args = {"path": path}
+            if mode:
+                args["mode"] = str(mode)
+            return self.cmd(cmd=cmd, args=args)
+
+    @error_context.context_aware
+    def file_close(self, handle):
```

Comment:
`file_close` and `file_flush`  are almost same, the rest are similar to each other as well.
I think you may probably  want to define a function like: `_file_operation` as base function for the rest to wrap it up , or `file_operation` that takes different instructions such as `open`, `close`... @luckyh  any suggestions?

Reply:
Yes, I know what you mean. Actually, all qga cmd function call a basic cmd(),you can see get_vcpus() and get_time() are almost the same.
About this patch,all functions are related to file, but I can't find a solution to define a basic function like file_operation and so on. I think the basic function is cmd().
So I want to leave them as they are.
Thanks.

Reply:
Yes, I know what you mean. Actually, all qga cmd function call a basic cmd(),you can see get_vcpus() and get_time() are almost the same.
About this patch,all functions are related to file, but I can't find a solution to define a basic function like file_operation and so on. I think the basic function is cmd().
So I want to leave them as they are.
Thanks.
@hereischen  I updated the other code,could you help to review?
The test results on rhel7 and rhel8 are updated in qemu_guest_agent: add test case for guest-file qga command #1399.

Reply:
Agree with @hereischen it could be something like `file_operation(action, **kwargs)`

Reply:
@vivianQizhu @hereischen  I updated it with file_operation(action, **kwargs), pls have a check.